downloader/src/main/kotlin/Main.kt

```diff
@@ -204,60 +201,82 @@ object Main {
                     e.printStackTrace()
                 }
 
-                // Clean up any files left from the failed VCS download (i.e. a ".git" directory)
-                targetDir.safeDeleteRecursively()
+                log.info { "VCS download failed for '${target.identifier}'." }
+
+                // Clean up any files left from the failed VCS download (i.e. a ".git" directory).
+                if (!targetDir.safeDeleteRecursively()) {
+                    log.warn { "Unable to cleanup ${targetDir.absolutePath} after failed VCS download." }
+                }
+
                 targetDir.safeMkdirs()
-                p("Download from VCS failed.")
             }
         }
 
-        p("Trying to download source package ...")
         return downloadSourcePackage(target, targetDir)
     }
 
     private fun downloadFromVcs(target: Package, outputDirectory: File): File {
-        val p = fun(string: String) = println("${target.identifier}: $string")
-
-        p("Trying to download from URL '${target.vcsProcessed.url}'...")
+        log.info {
+            "Trying to download '${target.identifier}' sources to '${outputDirectory.absolutePath}' from VCS..."
+        }
 
         if (target.vcsProcessed.url != target.vcs.url) {
-            p("URL was normalized, original URL was '${target.vcs.url}'.")
+            println("Using processed VCS URL '${target.vcsProcessed.url}'.")
+            println("Original VCS URL was '${target.vcs.url}'.")
+        } else {
+            println("Using VCS URL '${target.vcsProcessed.url}'.")
         }
 
         if (target.vcsProcessed.revision.isBlank()) {
-            p("WARNING: No VCS revision provided, downloaded source code does likely not match revision " +
-                    target.version)
+            log.warn {
+                "No VCS revision provided, downloaded source code will probably not match package version " +
+                        "${target.version}."
```

Comment:
No need to quote this.

Reply:
What about the trailing `.`? 😉 I *could* write

    + target.version + "."

if you find that more readable.

Reply:
No, I missed the dot.